src/DurableTask.Core/Tracing/TraceHelper.cs

```diff
@@ -357,6 +357,13 @@ internal static void EmitTraceActivityForSubOrchestrationFailed(
             OrchestrationInstance? instance,
             string? targetInstanceId)
         {
+
+            // We don't want to emit tracing for external events when they are related to entities
+            if ((targetInstanceId != null && targetInstanceId.StartsWith("@")) || (instance != null && instance.InstanceId.StartsWith("@")))
```

Comment:
I am not a fan of this approach of assuming any id that starts with `@` will have their events emitted elsewhere. There may be standalone customers not using entities with use `@` as a prefix. We shouldn't break their experience here.

Reply:
I tried implementing an alternative approach where I added a flag to  the `RequestMessage` that indicates whether or not we should create a trace. I then attempt to parse the `EventRaisedEvent.Input` into a `RequestMessage` here and check for this flag. The problem turned out to be that that we do not catch returns from a `CallEntity` request. These returns do not have type `RequestMessage` (they just look like `{result: 24}` for example) and as far as I can tell the only way to catch them is to check the format of the instance ID. It was my understanding that starting with "@" was reserved only for entities, and there seem to be examples of this assumption in the code already like [here](https://github.com/Azure/durabletask/blob/894674800424c4dc10a4b865949a018878b6cc72/src/DurableTask.Core/Common/Entities.cs#L30). But if you would still rather we not use this check, do you have any suggestions for alternative approaches?

Reply:
> it was my understanding that starting with "@" was reserved only for entities

Starting with `@` is only restricted to entities when entities are enabled, which is not by default. The history is that entities was only added in the past years, and using `@` for orchestrations has been supported long before then.

Reply:
For the purposes of unblocking this PR, I suggest we go ahead with Sophia's current approach. I agree that these assumptions already exist in many places, and we're not breaking any existing behavior with this particular change. We can certainly document it in our distributed tracing documentation, and I suggest we add a comment here too explaining the possibility of mis-labeling an orchestration as an entity.

@sophiatev rather than checking the string manually like this, let's leverage the API you linked to. Something like this:
```suggestion
            if (Entities.IsEntityInstance(targetInstanceId) || Entities.IsEntityInstance(instance.InstanceId))
```